### ISSUE TITLE
Disable more keyboard shortcuts if off in Settings

### ISSUE THREAD

**Issue by alex-key** [npmx-dev/npmx.dev#1779]
### Describe the bug

There are multiple places, where keyboard navigation still has custom implementation and disregards state in user settings:

- Search (ArrowUp, ArrowDown, Enter)
- Compage page (ArrowUp, ArrowDown, Enter)
- Package page > Playgrounds selector (ArrowUp, ArrowDown, Enter, Escape) [example](https://npmx.dev/package/@jctechstudio/ui)
- Package page > Package manager selector (ArrowUp, ArrowDown, Enter, Escape)
- Package page > Readme headers selector  (ArrowUp, ArrowDown, Enter, Escape)
- Package Code page > Version selector  (ArrowUp, ArrowDown, ArrowLeft, ArrowRight, Enter, Escape)

Need to disable ones that could be used with default keyboard navigation and decide about the rest which does not work w/o custom nav shortcuts

### Additional context

<img width="1794" height="882" alt="Image" src="https://github.com/user-attachments/assets/0e622d3e-938a-4534-8cb9-1d584cd4b7a5" />

<img width="1762" height="1430" alt="Image" src="https://github.com/user-attachments/assets/efd28fde-22bc-4fef-936a-56c310a01dde" />

<img width="2327" height="933" alt="Image" src="https://github.com/user-attachments/assets/817e86b2-40c1-498a-a9ce-f7fd75281c98" />

**Comment by alex-key** [comment]
@knowler Hi Nathan. I've found that some shorcuts still use custom implementation (listed in the issue above). I create a PR to disable it for first 3 listed (they will use user settings value). As for the rest: I checked and if custom navigation is disabled - default navigation by pressing "Tab" does not work. Could you please advice - how this should be addressed?

Default example - package manager selector on any package page: https://npmx.dev/package/vue#get-started

- focus on button in GET STARTED section
- press Space - dropdown list is opened
- press Tab
- focused button is "copy" instead of the first item from the list of package managers

**Comment by knowler** [comment]
@alex-key Anything that is implementing shortcuts according to the WAI-ARIA spec does not need to be disabled by the setting. This means any of the list boxes (e.g. package manager selector, README table of contents) and the combobox on the compare page can stay as those are the default shortcuts for them and they are what make those widgets operable by a keyboard. The WCAG requirement ([2.1.4 Character Key Shortcuts Level A](https://www.w3.org/WAI/WCAG22/quickref/#character-key-shortcuts)) is mainly that single character keys can be disabled or that they are limited to a single widget.

So, for the setting, I would only include the keyboard shortcuts currently listed in the “keyboard shortcuts” modal and nothing more.

**Comment by alex-key** [comment]
Would it be intuitive for user without any hints, that arrows should be used in such dropdowns instead of Tab?
Also is closing dropdown or modal pressing Escape allowed per WCAG?

**Comment by knowler** [comment]
@alex-key Yes, that is the expected affordance of those controls (compare with native select, input with suggestions, radio controls). It would make it break for assistive tech if we disabled these essential keyboard controls.

**Comment by alex-key** [comment]
@knowler Just to clarify: navigation in search results (which is not dropdowns) using arrows still should be disabled in Settings. Correct?

**Comment by knowler** [comment]
@alex-key Yep, those can be disabled since they’re actually shortcuts and not essential for navigating/operating.

**Comment by alex-key** [comment]
I will update PR accordingly
